action.sh

```diff
@@ -202,12 +202,22 @@ function start_vm {
 
   echo "The new GCE VM will be ${VM_ID}"
 
+  shutdown_command="gcloud compute instances delete $VM_ID --zone=$machine_zone --quiet"
+  startup_prelude="#!/bin/bash
+  set -e
+  shutdown() {
+    echo ❌ Machine setup failed so deleting $VM_ID in ${machine_zone} ...
+    ${shutdown_command}
+  }
+  trap shutdown EXIT
```

Comment:
@jder actually - wouldn't this run the shutdown as soon as the startup script finish - even if there was no problem?

Reply:
@jder please see https://github.com/related-sciences/gce-github-runner/commit/1ec4146438ff765ff4075a91936d69984391d5f9, let me know if that still works for you.

Reply:
Yeah, sorry, I was too focused on testing the failure case :) This makes more sense. Thanks!

Reply:
@jder no problem, btw our CI https://github.com/related-sciences/gce-github-runner/blob/008d5e1348dc6f84ab73b7f723b7fb67a9fc706d/.github/workflows/test.yml#L32 did not catch this because it's too quick (it's just an `echo`) - it was able to finish before the VM was shutdown.